### PR TITLE
fix(ui): select2-container

### PR DESCRIPTION
**Pull request by matthecarl**
Dropdown resizes properly when resizing the browser window. It uses `!important` to override jQuery inline style specificity.

Fixes GH-#6889

### CODE REVIEW

**Comment by PerStirpes** [comment]
I tried to avoid using the`!important` anti-pattern

* Bumping to the latest version: 4.0.6-rc.1 of select2 throws an exception 
`TypeError $(...).select2(...).select2(...).on is not a function`


* Tried manually setting the [container width](https://select2.org/appearance#container-width) configuration to `resolve` in a few places and broke all kinds of things. Happy to share more details if you like. 


Corrects “Default Role” drop-down inside settings. Doesn’t negatively affect the `stream tag filter`. It should also work properly for the “Country” drop-down: route organizations/<user>/billing/details/ 

I think using `!important` here avoids most of the common pitfalls that come with its use b/c if select2.js is ripped out I assume the styles associated with it would go too. 

@getsentry/ui

**Review by MaxBittker — APPROVED** [review]
based on the percy screenshots, i think that this looks like an improvement.

**Comment by MaxBittker** [comment]
@ckj I think this looks good to me but select2 is used a lot of place that may not be covered by snapshots, what are you thoughts on merging this?

**Comment by ckj** [comment]
@MaxBittker I'd confirm it's fine in the places we make the heaviest use of it... stream filtering, alert rule builder, etc. If those are okay, then I'd be comfortable 🚢'ing .

**Comment by MaxBittker** [comment]
good point, i'll stage & verify those places before 🚀. thanks again @PerStirpes ! sorry about the energy spent trying to bump select2, i've tried in the past and it's annoying.

**Comment by PerStirpes** [comment]
@ckj @MaxBittker 

Don't merge. I was looking for the alert rule builder page/feature and noticed the Audit log dropdown **/organizations/sentry/audit-log/** gets chopped & screwed. It has a ~250px fixed width and it looks bad when overridden. I'll take a look and maybe I can fix it w/o using !important. 

### **dev**
---
image


### **prod**
---
image

**Comment by MaxBittker** [comment]
thank you for checking 🙏

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR includes migrations</td>
    </tr>
  </tbody>
</table>

## Migration Checklist

- [ ] new columns need to be nullable (unless table is new)
- [ ] migration with any new index needs to be done concurrently
- [ ] data migrations should not be done inside a transaction
- [ ] before merging, check to make sure there aren't conflicting migration ids


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by PerStirpes** [comment]
😱   uh-oh

**Comment by PerStirpes** [comment]
What I did: I initially cloned the repo directly to my local machine(Friday), created a branch, worked on a fix, then forked the repo, then incorrectly set upstream to my fork, git pull origin master && git push upstream <branch>,  

sent pr, reviews were fine and changes were approved pending staging,  ckj noted a location I overlooked, I checked it and pr would introduce ui bug, made comment to not merge until I fixed it, 

This morning I git pull origin master and fixed ui issue this afternoon. And then I did this, 
 
image

afaik, I made four mistakes
1. Incorrectly setup origin and upstream 
2. I didn’t rebase on top of master
3. I didn’t write a test for the change I sent in
4. I’m super embarrassed because the worst part is I'm supposed to come to your office tomorrow for a job interview

**Comment by MaxBittker** [comment]
ha! no sweat @PerStirpes, this git thing is definitely solvable. to your knowledge is the patch working well besides that?

**Comment by PerStirpes** [comment]
Yes, but I can't see the billing page 👇 nor the percy snapshots so it's plausible that I missed something.

image

**Comment by MaxBittker** [comment]
Percy looks good, and we can check the billing page on staging and fix it downstream if need be, you don't need to worry about that as an open source contributor. As for the git situation, it might work to check out master, cherry pick your 2 commits from this branch, and then open a new PR (a renewable resource!) or force push to this one if you're confident it contains all of your changes

**Comment by razh** [comment]
FYI, this conflicts with the `width` option set in:

- [Select2Field](https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/components/forms/select2Field.jsx#L69)
- [SelectInput](https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/components/selectInput.jsx#L64)

In projects I work with, we generally end up setting `width: 100%` there.

**Comment by markstory** [comment]
Thank you for taking the time to put this together. Due to the number of conflicts and the reduced use of select2 across the application I'm going to close this.

**Pull request closed without merging**
